## Chapter 2 [Diffraction](CH2-_Diffraction.ipynb)


<hr style="height:1px;border-top:4px solid #FF8200" />

# The Electron 


part of 

## [Analysis of Transmission Electron Microscope Data](_Analysis_of_Transmission_Electron_Microscope_Data.ipynb)



by Gerd Duscher, 2019

Microscopy Facilities<br>
Joint Institute of Advanced Materials<br>
The University of Tennessee, Knoxville

Model based analysis and quantification of data acquired with transmission electron microscopes


## Non-relativistic de Broglie Wavelength

We generally use electron, photons, and neutrons for diffraction/scattering experiments.
These particles interact with differently with matter: 
- X-rays $<->$ electron density
- Neutrons $<->$ mass of nucleus
- Neutrons $<->$ magnetic moment
- Electrons $<->$ screened charge of nucleus

The electron is a elementary particle with spin $\frac{1}{2}$ (lepton).

Electrons can be viewed as particles and/or as waves.


Particle   -     wave dualism:

Scattering       $\leftrightarrow$ Diffraction

Particle picture $\leftrightarrow$ Wave picture


**Non--relativistic De Broglie wavelength** of electron: 

$\lambda = \frac{h}{p} = \frac{h}{\sqrt{2m_0E_{kin}}} \approx \frac{1.22}{\sqrt{E_{kin}}}$


E is the kinetic energy of the electron: $E_{kin} = eU $ [eV].

The wave length in a TEM is usually
a couple of picometers . This is a
factor of 100 smaller than your
XRD-source.



In [3]:
import numpy as np               #numeric library
import scipy.constants as const  #scientific constants

## input 
acceleration_voltage_V = U =200.0 *1000.0 #V   

## energy
E_kin = eV = const.e * acceleration_voltage_V  # potential

wave_length_m = const.h/np.sqrt(2*const.m_e*E_kin) # non-relativistic wavelength in m

##please note that we will keep all length units in nm if possible.
##otherwise we useonly SI units!!!
wave_length_nm = wave_length_m *1e9 # now in nm

print('Classic wave length is {0:.2f} pm for acceleration voltage {1:.1f} kV'.format(wave_length_nm*1000., acceleration_voltage_V/1000.))
# Notice that we change units in the output to make them most readable.

Classic wave length is 2.74 pm for acceleration voltage 200.0 kV


## Relativistic Correction
In the table above we see that the speeds of the electron is rather close to thespeed of light $c$

The formula for relativistic corrected wavelength is:
$\lambda = \frac{h}{\sqrt{2m_e E_{kin} *(1+\frac{E_{kin}}{2 m_e c^2})}}$

**Please note:** All units are internally in SI units: kg, s, V, J, except the length wihich is in nm!

We multiply with the appropriate factors for the output

In [4]:
# Input: Acceleration Voltage
E0 = acceleration_voltage = 20.0 *1000.0 #V

E_kin = eV = const.e * acceleration_voltage_V #potential

#relativisitic wavelength
wave_length = const.h/np.sqrt(2*const.m_e*E_kin*(1+E_kin/(2*const.m_e*const.c**2))) #in m

print(f'The relativistically corrected wave length is {wave_length*1e12:.2f} pm for acceleration voltage {acceleration_voltage/1000:.1f} kV')

The relativistically corrected wave length is 2.51 pm for acceleration voltage 20.0 kV


100kV : $\lambda$ = 4 pm $<$ than diameter an atom

The reltivistic parameters are:


|E (keV)|$\lambda$ (pm) | M/m$_0$ | v/c|
--------|---------------|---------|----|
|10 |  12.2 | 1.0796 | 0.1950 |
|30 | 6.98 | 1.129 | 0.3284 |
|100 | 3.70 | 1.1957 | 0.5482|
|200 | 2.51 | 1.3914 |  0.6953|
|400 | 1.64 | 1.7828 | 0.8275 |
|1000 | 0.87 | 2.9569 | 0.9411|

The same functionality (and code) is used in the KinsCat Library and we can test the values of above table.

Please change the acceleration voltage (**acceleration_voltage_V**) below.


## That means that the resolution is not limited by the wavelngth!

In [1]:
# Import Kinematic Scattering Library
import pyTEMlib.KinsCat as ks         # Kinematic sCattering Library

acceleration_voltage= 30*1e3
wave_length = ks.get_waveLength(acceleration_voltage)
print(f'The relativistically corrected wave length is {wave_length*1e3:.2f} pm for acceleration voltage {acceleration_voltage/1000:.1f} kV')

# Wavelength in 1/nm
def get_waveLength(E0):
    """
    Calculates the relativistic corrected de Broglie wave length of an electron

    Input:
    ------
        acceleration voltage in volt
    Output:
    -------
        wave length in 1/nm
    """

    eV = const.e * E0 
    return const.h/np.sqrt(2*const.m_e*eV*(1+eV/(2*const.m_e*const.c**2)))*10**9


Using KinsCat library version  0.5  by G.Duscher
spglib not installed; Symmetry functions of spglib disabled
windows
The relativistically corrected wave length is 6.98 pm for acceleration voltage 30.0 kV


In [4]:
help(ks.get_waveLength)

Help on function get_waveLength in module pyTEMlib.KinsCat:

get_waveLength(E0)
    Calculates the relativistic corrected de Broglie wave length of an electron
    
    Input:
    ------
        acceleration voltage in volt
    Output:
    -------
        wave length in nm



In [5]:
help(ks)

Help on module pyTEMlib.KinsCat in pyTEMlib:

NAME
    pyTEMlib.KinsCat

DESCRIPTION
    KinsCat
    Kinematic Scattering Theory
    Copyright by Gerd Duscher
    
    The Univerity of Tennessee, Knoxville
    Department of Materials Science & Engineering
    
    Sources:
               Scattering Theory:
               Zuo and Spence, "Advanced TEM", 2017
    
               Spence and Zuo, Electron Microdiffraction, Plenum 1992
      
               Atomic Form Factor:
               Kirkland: Advanced Computing in Electron Microsocpy 2nd edition
               Appendix C
    
    Units: everything is in SI units, except length is given in nm.
    
    Usage:
    All the input and ouptut is done through a dictionary
    (See Zuo_fig_3_18 for full example input)
    
    Available Subroutines:
    
        Kinematic_Scattering(tags, verbose=False ) # Main Routine
    
        CsCl()
        Silicon()
        Zuo_fig_3_18()
        
        feq(element, q) # atomic form factor at reci

## Particle Flux and Current

It is important todetermine the order of magitude of how many electrons are hitting the sample.

The electron sources deliver in the order of $\mu$A current, but most of these electrons are not used. 

In a modern electron microscope, we talk about a range of 1pA to 1nA in the electron beam.

We start with the defition of an Ampere:
$$A = \frac{C}{s}$$

That definition is enough to calculate the number ofelectron per time unit (flux).

In [6]:
print(f" elementary charge: {const.physical_constants['elementary charge'][0]:.5g} {const.physical_constants['elementary charge'][1]}")
print(f'\n 1pA is {1e-12/const.e:.3} electrons/s')
print(f'100pA is {100e-12/const.e*1 *1e-9:.3} electrons/ns')

print(f'\n at 15pA an electron will hit the sample every {const.e/15e-12 * 1e9:.2f} ns ')


 elementary charge: 1.6022e-19 C

 1pA is 6.24e+06 electrons/s
100pA is 0.624 electrons/ns

 at 15pA an electron will hit the sample every 10.68 ns 


We see that we have much lower fluence in the TEM than in a laser (how could they do femtosecond pulses otherwise).


## Back: Chapter 1: [Introduction](CH1-Introduction.ipynb)
## Next: [Atomic Form Factor](CH2-Atomic_Form_Factor.ipynb)
## Chapter 2: [Diffraction](CH2_Diffraction.ipynb)
## List of Content: [Front](_Analysis_of_Scanning_Electron_Microscope_Data.ipynb)
